commands/python/command/deploy.py

```diff
@@ -56,7 +57,7 @@ def deploy(project, git_ref, namespace, dry_run=False, force=False, force_helm=F
             logging.error(f'You are deploying hash {git_ref} which does not match branch '
                           f'{cr.check_branch} on cluster {cr.cluster_name} for project '
                           f'{project}... exiting')
-            return
+            sys.exit(1)
```

Comment:
I think the command dispatch module ought to interpret non-0 or non-None return values as failures and exit the overall command in such a way that the calling shell gets an error exit code